### DISCUSSION TITLE
Wiki Pages

### DISCUSSION THREAD

**Discussion by Sn1kk3r5** [JohanDegraeve/xdripswift#204] [Q&A]
Hi Johan,

I wonder if there’s a possibility  to support the project by updating the wiki pages. 
Some information are copied over and outdated. 
Can this be done via forking the project as well?

Cheers
Cris

**Comment by JohanDegraeve** [comment]
maybe this can help ?
https://stackoverflow.com/questions/7197937/on-github-can-i-fork-just-a-wiki

**Reply by Sn1kk3r5** [reply]
I‘ll give it a try. 

Sorry I could have thought about this topic before and research myself. 

Will keep you posted

**Reply by JohanDegraeve** [reply]
no problem, Ok I'll hear from you

**Reply by Sn1kk3r5** [reply]
Well, got it so far.

You can't edit a wiki directly within the browser. You need to have at least a local editor in place.
GitHub Desktop doesn't work, but Visual Studio Code does. --> on the wiki site down on the left corner there is direct link available to clone the wiki only.

So for now I was able to clone the pages and can edit them offline.
How the push will work I need to figure out later.

Cheers
Cris

**Reply by Sn1kk3r5** [reply]
Hi Johan,

what I investigated, it's not possible to pull a wiki back to it's origin.
Do you have any suggestions to handle this now?

Or do you say, I got enough to do's currently and can't effort the time for a wiki?

Cheers
Cris

**Comment by JohanDegraeve** [comment]
Sorry that should be for you
git clone https://github.com/Sn1kk3r5/xdripswift.wiki.git

**Reply by JohanDegraeve** [reply]
sorry it seams you can not fork the WiKi seperately ?

**Reply by Sn1kk3r5** [reply]
The fork itself is working with VS Code, but I can't pull it after the changes.
From what I investigated, none of the provided possible solution are working. 👎

**Comment by JohanDegraeve** [comment]
you say "pull" ? I assume you mean "push"

**Reply by Sn1kk3r5** [reply]
Typo, push of course

**Reply by JohanDegraeve** [reply]
strange that it works for me but not for you

can you try this in a Terminal session ?

cd to the directory where you cloned the wiki
git status
==> should give a list of changed files 
git add .
==> adds all changes files for commit
git commit -m "your message"
==> commits them
git push
==> pushes to your fork

**Reply by Sn1kk3r5** [reply]
same as before: error saying
Username for 'https://github.com': Sn1kk3r5
Password for 'https://Sn1kk3r5@github.com': 
remote: Permission to JohanDegraeve/xdripswift.wiki.git denied to Sn1kk3r5.
fatal: unable to access 'https://github.com/JohanDegraeve/xdripswift.wiki.git/': The requested URL returned error: 403

**Reply by Sn1kk3r5** [reply]
OK got an email from GitHUb saying, basic authentication is not supported anymore....
I need to see if I can avoid this issue somehow with VS Code.

**Reply by JohanDegraeve** [reply]
ah ok,
so the remote is my repository, it's normal you can't push to it

the real issue is you can't fork the wiki

I found this in stackoverflow
- Fork a repo. The wiki is not forked automatically. So your fork will have empty wiki.
- Create a wiki in the forked repo. Just create an empty Home page or whatever.
- Clone the git repo of original wiki: git clone https://github.com/JohanDegraeve/xdripswift.wiki.git
- Assuming your forked repo now sits in https://github.com/Sn1kk3r5/xdripswift, add the new remote to the wiki repo: git remote add fork https://github.com/Sn1kk3r5/xdripswift.wiki.git
- Do the force push to overwrite your wiki with the original git push -f fork master

https://stackoverflow.com/questions/7197937/on-github-can-i-fork-just-a-wiki

**Reply by Sn1kk3r5** [reply]
I was able to folk the repo by using the link in the right corner of the origin wiki.
I do ave a local copy (clone) of the existing wiki in VS Code.

It'S the push only witch doesn't work.

**Reply by Sn1kk3r5** [reply]
Can you see my latest push.

Push went trough without any notification

**Reply by JohanDegraeve** [reply]
yes in your fork, which is ok https://github.com/Sn1kk3r5/xdripswift/wiki

but now you would need to get my content in your fork
then you can do changes, push them to your fork, then I should be able to set your fork as a remote and cherry-pick your commit(s)

next steps
- go to a terminal session and cd to a folder where you have your fork cloned
- git remote add remote_johan https://github.com/JohanDegraeve/xdripswift.wiki.git
This allows you to fetch content from my repo into yours
- get fetch remote_johan
This does the actual fetch
to copy the contents from the remote to yours:
- get merge remote_johan/master

but I think you will get a morge conflict, because you've now updated Home.md

maybe you'l need to restart from scratch
- delete your local clone
- clone again
- then do steps above

**Reply by Sn1kk3r5** [reply]
This is so wired.
Regarding to VS Code, the push went through, but didn't.
I do have the recreated sites still locally, so no much lost :-)

What I understand is.

I create a wiki in my repo based on your wiki in your repo.
Then I'll do the changes in my wiki and you cherry pick them. 

Correct?

**Reply by JohanDegraeve** [reply]
yes that's it
you can't push to my repo, so you create a fork
but the fork doesn't copy my contents
You get my contents by configuring my repo as remote to your fork
then you fetch my contents
you push my contents to your fork
then you start applying changes

once done I can do the opposit
I set your repo as my remote
and then I get your commits

if that works, then we need to make sure we don't update the same files simulateneously, otherwise we get merge conflicts

**Reply by Sn1kk3r5** [reply]
Good, I'll give this a try later today.

I'm sure we'll find a way to avoid double work ;-)

**Reply by Sn1kk3r5** [reply]
for what ever reason I din't get the remote to work.

I was able to get your repos wiki to my local hdd, did changes and could upload them to my repos wiki.

But this was only possible by copying the files over from my local repo to your local repo.

I assume this is not how it should work. I don't even no if the local repo on my hdd will recieve a fetch in case you did change something yourself to the wiki.

Never the less I did't made an example page for the Apple watch and renamed the page title.

**Comment by JohanDegraeve** [comment]
this looks ok, and I think this can work
I see your changes here https://github.com/Sn1kk3r5/wiki_test_for_xDrip4iOS/wiki
Once you give me a go I can review them and merge your commit into my repository

**Reply by Sn1kk3r5** [reply]
OK, I'll get back to you when I did my changes.
Thanks for your help and support!

**Reply by Sn1kk3r5** [reply]
Hi Johan,

can we give this a try? I did some changes e.g. _sidebar.md got a new section and renamed some displayed page titles.
Apple-Watch.md page is completely renewed and Bug-Reporting.md is a whole new page.

**Links in sidebar.md are pointing to your repos wiki!**

**Reply by JohanDegraeve** [reply]
this looks good. It's a pitty I can't see the changes as with code
I'll see if there other ways to achieve this, maybe in sourcetree

You continue making updates ?

**Reply by Sn1kk3r5** [reply]
Currently not as I want to ensure if my local repo will be updated once you added the changes. 

If my local repo won‘t recognize the change on your repos wiki, I need to investigate a new way as the manual way takes to much time and is eligible for errors and mismatches. 

Is there a possibility to give editing authority to contributors?

**Reply by JohanDegraeve** [reply]
No I will not give editing autority

It's indeed taking much time and I failed so far to merge your changes.

Probably it's better that you don't start from a fork but from a new project 

in a terminal session 
git init
git remote add JohanWiki https://github.com/JohanDegraeve/xdripswift.wiki.git
git pull JohanWiki master
git merge JohanWiki/master

now you should have an onw repository based on my repository
do your changes
git add .
git commit -m "your commit message"

now you would need to push this new project in github, in your account
probably something like 
git remote add origin https://github.com/Sn1kk3r5/newprojectname.wiki.git ==> not sure if that works, maybe your first need to create a new/empty project on github
then 
git push (possible git push --force)

**Comment by Sn1kk3r5** [comment]


**Reply by JohanDegraeve** [reply]
you're right, 
alternative is that you edit the files and send them to me ? (xdrip@proximus.be)

**Reply by Sn1kk3r5** [reply]
well that's a possibility, but also an overhead.

Clone your wiki to my local hdd...
Do changes
Upload to my repo.
Only restricted possibility of testing (Sidebar must point to your repos wiki)
packing all files and send via email....

And then it's on your side.

Quit a lot of work for a wiki.
